### PR TITLE
Update AB2 (mpas_ocn_time_integration_split_ab2.F) code

### PR DESCRIPTION
**Pull request by hyungyukang**
This PR updates AB2 time stepping code.

Some codes and a routine were updated in `split.F`, but not in `split_ab2.F`

* `ocn_diagnostic_solve_wctEdge` routine was updated, but `split_ab2.F` was using a call sentence of the old version of the routine.
   - `deltaSSH` computation was missing in `split_ab2.F`.
   - `baroclinicThickness` is computed instead of 'bottomDepthEdge'.
* 'Direct application of tidal boundary condition', which was only in `split.F`, is added to `split_ab2.F`.

### CODE REVIEW

**Comment by hyungyukang** [comment]
Passed all COMPASS nightly suite on Perlmutter with GNU compiler.
```
Test Runtimes:
00:49 PASS ocean_baroclinic_channel_10km_default
01:38 PASS ocean_baroclinic_channel_10km_threads_test
00:44 PASS ocean_baroclinic_channel_10km_decomp_test
00:27 PASS ocean_baroclinic_channel_10km_restart_test
28:03 PASS ocean_global_ocean_QU240_mesh
01:56 PASS ocean_global_ocean_QU240_WOA23_init
00:55 PASS ocean_global_ocean_QU240_WOA23_performance_test
02:30 PASS ocean_global_ocean_QU240_WOA23_restart_test
02:02 PASS ocean_global_ocean_QU240_WOA23_decomp_test
02:04 PASS ocean_global_ocean_QU240_WOA23_threads_test
01:37 PASS ocean_global_ocean_QU240_WOA23_analysis_test
01:13 PASS ocean_global_ocean_QU240_WOA23_RK4_performance_test
01:59 PASS ocean_global_ocean_QU240_WOA23_RK4_restart_test
01:55 PASS ocean_global_ocean_QU240_WOA23_RK4_decomp_test
03:12 PASS ocean_global_ocean_QU240_WOA23_RK4_threads_test
00:00 PASS ocean_global_ocean_QUwISC240_mesh
00:00 PASS ocean_global_ocean_QUwISC240_WOA23_init
03:00 PASS ocean_global_ocean_QUwISC240_WOA23_performance_test
00:41 PASS ocean_ice_shelf_2d_5km_z-star_restart_test
00:23 PASS ocean_ice_shelf_2d_5km_z-level_restart_test
00:21 PASS ocean_ziso_20km_default
00:27 PASS ocean_ziso_20km_with_frazil
Total runtime 56:00
PASS: All passed successfully!
```

File: components/mpas-ocean/src/mode_forward/mpas_ocn_time_integration_split_ab2.F [hunk 1]
```diff
@@ -1187,8 +1216,16 @@ subroutine ocn_time_integrator_split_ab2(domain, dt)!{{{
...
+            !$omp parallel
+            !$omp do schedule(runtime)
+            do iCell = 1, nCellsAll
+               deltaSSH(iCell) = sshSubcycleCur(iCell) - sshNew(iCell)
+            end do
+            !$omp end do
+            !$omp end parallel
+
+            call ocn_diagnostic_solve_wctEdge(btrvel_temp, deltaSSH, &
+                      baroclinicThickness, wctEdge)
```
Comment (mark-petersen) [inline components/mpas-ocean/src/mode_forward/mpas_ocn_time_integration_split_ab2.F:1228]: @hyungyukang commented in a meeting that this is the critical change in this PR.

Comment (cbegeman) [reply components/mpas-ocean/src/mode_forward/mpas_ocn_time_integration_split_ab2.F:1228]: It looks from https://github.com/E3SM-Project/E3SM/pull/6047 that this would be BFB when the thickness is centered, so I don't think this will affect the instability we encountered in the v3 HR G-case.

**Comment by mark-petersen** [comment]
@hyungyukang, it would be better to separate this PR into two PRs: one that is BFB on all tests with the majority of changes, and one with the remaining non-BFB lines. That way it is clear what code changes are non-BFB. It looks like many of these changes are clean-up and conforming to the split-explicit code, but are not changing the computation. It is very helpful to have non-BFB PRs with small isolated changes, even just a few lines, when we look back later and ask how it changed the results.

**Comment by hyungyukang** [comment]
> @hyungyukang, it would be better to separate this PR into two PRs: one that is BFB on all tests with the majority of changes, and one with the remaining non-BFB lines. That way it is clear what code changes are non-BFB. It looks like many of these changes are clean-up and conforming to the split-explicit code, but are not changing the computation. It is very helpful to have non-BFB PRs with small isolated changes, even just a few lines, when we look back later and ask how it changed the results.

@mark-petersen , Thanks for suggesting. Agreed. I will close this PR first, then open one PR with BFB on all tests. Maybe it'd be better to open another PR after merging the BFB PR. Thanks!

**Comment by hyungyukang** [comment]
Closing this PR and splitting it into two PRs to clarify the BFB/non-BFB changes as suggested by a reviewer.
